src/flamegraph/flamegraph.js

```diff
@@ -198,41 +199,42 @@ function update_text(e) {
 // zoom
 function zoom_reset(e) {
     if (e.attributes != undefined) {
-        orig_load(e, "x");
-        orig_load(e, "width");
+        if (e.attributes.x_samples != undefined) {
```

Comment:
Why would this ever be `undefined`?

Reply:
`zoom_reset` is called on every child of `frames`. It's not just passed the rectangles, but also the text and the parent that wraps the rectangle + text. I just learned this in answering your question, and I've tried to clarify the code.